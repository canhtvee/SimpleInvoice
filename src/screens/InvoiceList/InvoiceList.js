import React from 'react';
import {View} from 'react-native';
import {FormProvider, useForm} from 'react-hook-form';

export function InvoiceList() {
  const form = useForm({mode: 'onSubmit', reValidateMode: 'onChange'});

  return (
    <FormProvider {...form}>
      <View style={{flex: 1}}></View>
    </FormProvider>
  );
}
