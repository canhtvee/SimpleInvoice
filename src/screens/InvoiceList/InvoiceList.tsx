import React from 'react';
import {Text, View} from 'react-native';

export function InvoiceList({navigation}: any) {
  return (
    <View>
      <Text onPress={() => navigation.navigate('CreateInvoice')}>
        InvoiceList
      </Text>
    </View>
  );
}
