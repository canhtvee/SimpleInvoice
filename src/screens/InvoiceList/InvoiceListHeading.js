import React from 'react';
import {ScrollView, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {useFormContext} from 'react-hook-form';

import {AppButton} from '../../commons';

export function InvoiceListHeading() {
  const navigation = useNavigation();
  const {setValue} = useFormContext();

  return (
    <View>
      <ScrollView horizontal showsHorizontalScrollIndicator={false}>
        <AppButton
          title="Create Invoice"
          onPress={() => navigation.navigate('CreateInvoice')}
          textButton
        />
        <AppButton
          title="Sort by Latest"
          textButton
          onPress={() => setValue('sortOrder', 'LATEST')}
        />
        <AppButton
          title="Sort by Newest"
          textButton
          onPress={() => setValue('sortOrder', 'NEWEST')}
        />
      </ScrollView>
    </View>
  );
}
