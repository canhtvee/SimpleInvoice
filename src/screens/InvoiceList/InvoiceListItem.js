import React from 'react';
import {Pressable, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';

import {AppText} from '../../commons';
import {Colors, Sizes} from '../../utils';

function InvoiceListItemField({label, value}) {
  return (
    <View style={{flexDirection: 'row'}}>
      <AppText style={{width: Sizes.wpx(100)}}>{label}:</AppText>
      <AppText style={{overflow: 'hidden', width: Sizes.wpx(240)}}>
        {value}
      </AppText>
    </View>
  );
}

export function InvoiceListItem({data, index}) {
  const navigation = useNavigation();

  return (
    <Pressable
      style={{
        marginVertical: Sizes.paddinglxx,
        marginHorizontal: Sizes.padding,
        alignItems: 'center',
        borderRadius: Sizes.borderRadius,
        backgroundColor: Colors.background,
        padding: Sizes.paddinglx,
      }}
      onPress={() => {
        navigation.navigate('InvoiceDetail', {invoice: data});
      }}>
      <InvoiceListItemField label={'Reference'} value={data.invoiceReference} />
      <InvoiceListItemField label={'Due Date'} value={data.dueDate} />
      <InvoiceListItemField label={'Amount'} value={data.amount} />
      <InvoiceListItemField label={'Description'} value={data.description} />
    </Pressable>
  );
}
