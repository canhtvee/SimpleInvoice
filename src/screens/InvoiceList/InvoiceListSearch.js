import React, {useEffect, useRef, useState} from 'react';
import {TextInput} from 'react-native';

import {useFormContext} from 'react-hook-form';
import {Colors, CommonStyles, Sizes} from '../../utils';

export function InvoiceListSearch() {
  const {setValue} = useFormContext();
  const timer = useRef();
  const [state, setState] = useState();

  useEffect(() => {
    timer.current && clearTimeout(timer.current);

    timer.current = setTimeout(() => setValue('searchTerm', state), 1000);
  }, [state]);

  useEffect(() => {
    return () => timer.current && clearTimeout(timer.current);
  }, []);

  return (
    <TextInput
      style={[
        {
          fontSize: Sizes.regular,
          paddingHorizontal: Sizes.paddinglx,
          borderColor: Colors.border,
          borderWidth: 1,
          borderRadius: Sizes.borderRadius,
          margin: Sizes.padding,
        },
        CommonStyles.textInputPadding,
      ]}
      placeholder={'Search Invoice By DueDate'}
      value={state}
      onChangeText={text => setState(text)}
      clearButtonMode={'always'}
    />
  );
}
