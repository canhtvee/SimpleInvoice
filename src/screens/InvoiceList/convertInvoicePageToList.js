export const convertInvoicePageToList = pages => {
  const invoiceList = [];
  pages?.forEach(page => {
    invoiceList.push(...page.data);
  });

  return invoiceList;
};
