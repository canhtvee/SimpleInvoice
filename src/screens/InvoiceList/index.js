import {InvoiceService} from './modules';

export {InvoiceService};

export * from './InvoiceList';
