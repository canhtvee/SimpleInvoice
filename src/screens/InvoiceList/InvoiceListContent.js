import React from 'react';
import {FlatList} from 'react-native';

import {AppContentEmpty} from '../../commons';
import {Colors, Sizes} from '../../utils';

import {useInvoiceList} from './modules';
import {InvoiceListItem} from './InvoiceListItem';

export function InvoiceListContent() {
  const invoiceList = useInvoiceList();

  return (
    <FlatList
      data={invoiceList}
      contentContainerStyle={{
        paddingBottom: Sizes.padding * 2,
        backgroundColor: Colors.ripple,
        paddingTop: Sizes.paddinglxx,
      }}
      renderItem={({item, index}) => (
        <InvoiceListItem data={item} index={index} />
      )}
      keyExtractor={(item, index) => `${item.invoiceId}-${index}`}
      ListEmptyComponent={
        <AppContentEmpty containerStyle={{marginTop: Sizes.space3x}} />
      }
    />
  );
}
