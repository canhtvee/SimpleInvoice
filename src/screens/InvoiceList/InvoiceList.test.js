import React from 'react';
import {cleanup} from '@testing-library/react-native';

import {renderWithNavigator} from '../../utils';
import {InvoiceList} from './InvoiceList';

/**
 * Run test command
 * yarn test -u -t="InvoiceList"
 */

afterEach(cleanup);

describe('InvoiceList', () => {
  test('should render correctly', () => {
    const tree = renderWithNavigator(<InvoiceList />);

    expect(tree).toMatchSnapshot();
  });
});
