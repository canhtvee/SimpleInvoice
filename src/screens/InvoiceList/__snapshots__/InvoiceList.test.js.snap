// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceList should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    [
      "left",
      "right",
    ]
  }
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <TextInput
    clearButtonMode="always"
    onChangeText={[Function]}
    placeholder="Search Invoice By DueDate"
    placeholderTextColor="#00000061"
    style={
      [
        {
          "borderColor": "#0000008A",
          "borderRadius": 6,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 27.5,
          "margin": 23.5,
          "paddingHorizontal": 16,
        },
        {
          "paddingHorizontal": 16,
          "paddingVertical": 23.5,
        },
      ]
    }
  />
  <View>
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={16}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 6,
              "height": 79,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 47,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 29.5,
                  "fontWeight": "600",
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "color": "#2196f3",
                },
                undefined,
              ]
            }
          >
            Create Invoice
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={16}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 6,
              "height": 79,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 47,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 29.5,
                  "fontWeight": "600",
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "color": "#2196f3",
                },
                undefined,
              ]
            }
          >
            Sort by Latest
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={16}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 6,
              "height": 79,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 47,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 29.5,
                  "fontWeight": "600",
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "color": "#2196f3",
                },
                undefined,
              ]
            }
          >
            Sort by Newest
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <RCTScrollView
    ListEmptyComponent={
      <AppContentEmpty
        containerStyle={
          {
            "marginTop": 59,
          }
        }
      />
    }
    contentContainerStyle={
      {
        "backgroundColor": "#0000001A",
        "paddingBottom": 47,
        "paddingTop": 8,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "marginTop": 118.5,
              "width": "100%",
            },
            {
              "marginTop": 59,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontSize": 27.5,
              },
              [
                {
                  "paddingHorizontal": 23.5,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
        >
          Data is not available
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
